libraries/matrix/api/src/main/kotlin/io/element/android/libraries/matrix/api/verification/SessionVerificationService.kt

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2023 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.libraries.matrix.api.verification
+
+import kotlinx.coroutines.flow.StateFlow
+
+interface SessionVerificationService {
+
+    val verificationAttemptStatus : StateFlow<SessionVerificationServiceState>
+
+    /**
+     * The internal service that checks verification can only run after the initial sync.
+     * This [StateFlow] will notify consumers when the service is ready to be used.
+     */
+    val isReady: StateFlow<Boolean>
+
+    /**
+     * Exposes whether the current session is verified or not.
+     */
+    val isVerified: StateFlow<Boolean>
+
+    /**
+     * Request verification of the current session.
+     */
+    fun requestVerification()
```

Comment:
Can all these methods fail? if yes, maybe returns `Result<Unit>`

Reply:
With the current approach, if any operation fails the error will be passed to the StateFlow as `SessionVerificationServiceState.Failed` and displayed as such in the View. I think that's better than having local errors (Rust panics when calling those methods) and actual flow failures handled in a different way?

Reply:
Yes it's ok I guess